src/go/rpk/pkg/config/params.go

```diff
@@ -52,18 +52,18 @@ const (
 	// This block contains names X flags that are used for backcompat.
 	// All new X flags are defined directly into the xflags slice.
 	xKafkaBrokers       = "brokers"
-	xKafkaTLSEnabled    = "brokers.tls.enabled"
-	xKafkaCACert        = "brokers.tls.ca_cert_path"
-	xKafkaClientCert    = "brokers.tls.client_cert_path"
-	xKafkaClientKey     = "brokers.tls.client_key_path"
-	xKafkaSASLMechanism = "brokers.sasl.mechanism"
-	xKafkaSASLUser      = "brokers.sasl.user"
-	xKafkaSASLPass      = "brokers.sasl.pass"
+	xKafkaTLSEnabled    = "tls.enabled"
+	xKafkaCACert        = "tls.ca"
+	xKafkaClientCert    = "tls.cert"
+	xKafkaClientKey     = "tls.key"
+	xKafkaSASLMechanism = "sasl.type"
```

Comment:
I went back and forth on this. I prefer mechanism too, but the yaml files currently use `type` rather than `mechanism` for the field name and I figured to match them for simplicity.

I remember when I first learned SASL, I didn't really understand what "mechanism" meant or was. I think I prefer mechanism now, but I wonder what's easier for newcomers...

Options:
(a) keep this as is to match the yaml file
(b) change the yaml file to decode `type`, but always encode `mechanism` -- similar to how I'm renaming rpk's `truststore_file` to `ca_file` in this PR
(c) change this to sasl.mechanism, keep the yaml file as `type`

wdyt?

Reply:
For posterity: we discussed and are proceeding with (b)